### PR TITLE
Upload API unit test results to dashboard

### PR DESCRIPTION
**Pull request by jessicamack**
##### SUMMARY

Upload unit test results to dashboard.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -97,6 +97,22 @@ jobs:
             }}
           token: ${{ secrets.CODECOV_TOKEN }}
 
+      - name: Upload awx jUnit test reports
+        if: >-
+          matrix.test.name == 'api-test'
```
Comment (webknjaz) [inline .github/workflows/ci.yml]: That check is not actually necessary, I dropped it from other jobs.
```suggestion
          !cancelled()
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -97,6 +97,22 @@ jobs:
...
+          && steps.make-run.outputs.test-result-files != ''
+          && github.event_name == 'push'
+          && github.repository == 'ansible/awx'
```
Comment (webknjaz) [inline .github/workflows/ci.yml]: This is more reliable and how I've implemented the upstream check above.
```suggestion
          && env.UPSTREAM_REPOSITORY_ID == github.repository_id
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -97,6 +97,22 @@ jobs:
...
+          && github.ref_name == 'devel'
```
Comment (webknjaz) [inline .github/workflows/ci.yml]: This is more reusable:
```suggestion
          && github.ref_name == github.event.repository.default_branch
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -97,6 +97,22 @@ jobs:
...
+        continue-on-error: true
```
Comment (webknjaz) [inline .github/workflows/ci.yml]: I thought we agreed this is unnecessary?
```suggestion
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -97,6 +97,22 @@ jobs:
...
+        run: >-
+          curl -v --user "${{ vars.PDE_ORG_RESULTS_AGGREGATOR_UPLOAD_USER }}:${{ secrets.PDE_ORG_RESULTS_UPLOAD_PASSWORD }}"
+          --form "xunit_xml=${{ steps.make-run.outputs.test-result-files }}"
```
Comment (webknjaz) [inline .github/workflows/ci.yml]: Note that `${{ steps.make-run.outputs.test-result-files }}` may contain several files, so you probably need multiple `curl` calls unless the endpoint accepts multiple files in a single playload.

**Review by webknjaz — COMMENTED** [review]
Additionally, this upload step needs to be added to other jobs as they also produce junit files due to recently merged PRs.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (webknjaz) [reply .github/workflows/ci.yml]: Also, add a leading `@`

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -97,6 +97,22 @@ jobs:
...
+          --form "component_name=awx"
+          --form "git_commit_sha=${{ github.sha }}"
+          --form "git_repository_url=https://github.com/${{ github.repository }}"
+          "${{ vars.PDE_ORG_RESULTS_AGGREGATOR_UPLOAD_URL }}/api/results/upload/"
```
Comment (webknjaz) [inline .github/workflows/ci.yml]:
```suggestion
        run: |
          for junit_file in $(echo '${{ steps.make-run.outputs.test-result-files }}' | sed 's/,/ /')
          do
              curl \
                -v \
                --user "${{ vars.PDE_ORG_RESULTS_AGGREGATOR_UPLOAD_USER }}:${{ secrets.PDE_ORG_RESULTS_UPLOAD_PASSWORD }}"
                --form "xunit_xml=@${junit_file}" \
                --form "component_name=awx"
                --form "git_commit_sha=${{ github.sha }}"
                --form "git_repository_url=https://github.com/${{ github.repository }}"
                "${{ vars.PDE_ORG_RESULTS_AGGREGATOR_UPLOAD_URL }}/api/results/upload/"
          done
```

**Review by chrismeyersfsu — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -97,6 +97,26 @@ jobs:
...
+        if: >-
+          !cancelled()
+          && steps.make-run.outputs.test-result-files != ''
+          && github.event_name == 'push'
+          && env.UPSTREAM_REPOSITORY_ID == github.repository_id
+          && github.ref_name == github.event.repository.default_branch
+        run: |
+          for junit_file in $(echo '${{ steps.make-run.outputs.test-result-files }}' | sed 's/,/ /')
+          do
+              curl \
```
Comment (chrismeyersfsu) [inline .github/workflows/ci.yml:110]: I think @webknjaz mentioned a leading @ here.

**Pull request merged**